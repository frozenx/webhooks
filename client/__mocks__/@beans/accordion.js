

import React from 'react';

const Accordion = (props) => (<div {...props} />);

export default { Accordion };