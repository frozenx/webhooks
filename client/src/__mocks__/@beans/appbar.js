import React from 'react';
const AppBar = (props) => (<div {...props}/>);

export default AppBar;
