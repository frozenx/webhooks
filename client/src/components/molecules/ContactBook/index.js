import React, { Fragment } from "react";
import { FormHeading, PrimaryWrapper, SecondaryWrapper, FormHeadingWrapper } from "../../../common/styles";
import Button from '@beans/button';
import BackComponent from '../SupplierForm/BackComponent'
import CardListWithPagination from '../ContactCardList'
import i18Data from '../../../languagepack';

const ContactBook = ({
    showContactForm,
    contactList,
    getNewPage,
    partnerName,
    handleBackComponent,
    navigateToForm,
    showLoader,
    totalCount,
    currentPage,
    handlePageChange

}) => {
    return (
        <Fragment>
            <BackComponent handleBackComponent={handleBackComponent} />
            <PrimaryWrapper>
                <SecondaryWrapper>
                    <FormHeadingWrapper>
                        <FormHeading>
                            <h1>
                                {`${partnerName} ${i18Data.contacts}`}
                            </h1>
                        </FormHeading>
                        <Button onClick={showContactForm} variant='secondary'>
                            {i18Data.addNewContact}
                        </Button>
                    </FormHeadingWrapper>
                    <CardListWithPagination 
                        paginationProps = {
                            {
                                totalCount,
                                getNewPage, 
                                linkTemplate : "/page=%{pageNumber}",
                                pageText : "Page",
                                currentPage, 
                                handlePageChange
                            }
                        }
                        componentProps = {
                            {
                                contactList,
                                showLoader,
                                navigateToForm,
                                handleBackComponent

                            }
                        }
                    />
                </SecondaryWrapper>
            </PrimaryWrapper>
        </Fragment>
    );
}


export default ContactBook;