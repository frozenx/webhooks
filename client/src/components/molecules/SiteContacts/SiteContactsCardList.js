import React, { Fragment } from 'react';
import { FormHeading, FormHeadingWrapper, StyleGrayWrapper } from "../../../common/styles";
import Label from '@beans/label';
import Button from '@beans/button';
import i18Data from '../../../languagepack';
import AdministrativeContactCard from './AdminstrativeContactCard';

import {
    CardTopContainer,
    CardBottomContainer,
    StyledLeftSection,
    StyledRightSection,
    IconStyle
} from '../../../common/styles'
import StatusIndicator from '@beans/colleague-status-indicator';
import Dropdown from '@beans/dropdown';
import Icon from '@beans/icon';
import { DropdownWrapper } from './styled'


export default class SiteContactsCardList extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            contactTypes: {
                billToContact: {},
                shipToContact: {},
                correspondenceContact: {},
                dunningContact: {},
                InvoicesContact: {}
            },
            isContactTypesPresent: false,
            editContact: false,
            editContactStatus: false,
            showLoader: false
        }
    }

    getContact = (uuid) => {                
        let foundContact = this.props.rootNode.contactBook.getContact(uuid)            
        if(foundContact) {
            let { contactStatus, firstName, lastName, contactUuid } = foundContact;
            let contact = { contactStatus, firstName, lastName, contactUuid }
            return contact
        }
        return {};
    }

    componentDidMount() {
        let allContactTypes = this.props.node.contact.values
        let savedContactTypes = {...allContactTypes}
        delete savedContactTypes.administrativeContact;
        let contactTypes = { ...this.state.contactTypes }
        let isSavedContactTypesEmpty = Object.keys(allContactTypes).every((contactType) => !allContactTypes[contactType])                
        if (!isSavedContactTypesEmpty) {
            Object.keys(savedContactTypes).map((contact, i) => {
                if (savedContactTypes[contact]) {
                    contactTypes[contact] = this.getContact(savedContactTypes[contact]);
                }
                this.setState({
                    contactTypes,
                    isContactTypesPresent: true
                })
            })
              
        }

        else {           
            this.setState({
                isContactTypesPresent: false
            })

        }            
    }
    onChange = (contactType, e) => {
        let foundContact = this.getContact(e.target.value)
        let contactTypes = {...this.state.contactTypes}
        contactTypes[contactType.id] = foundContact;
        this.setState({
            contactTypes,
            isContactTypesPresent: true
        }, () => {
            this.props.editSiteContacts(false)
            this.props.siteContactData(this.state.contactTypes, false);
        })
    }

    updateContact = (contactType) => {
        let contactTypes = { ...this.state.contactTypes }       
        contactTypes[contactType.id] = {}
        this.setState({
            contactTypes
        }, () => {
            this.props.siteContactData(this.state.contactTypes, false)
        }
        );
    }
    updateEditableStatus = (status) =>{        
        this.setState({
            isContactTypesPresent: status
        })
    }

    render() {
        const {
            contacts,
            contactList,
            rootNode,
            node,
            editSiteContacts,
            iseditSiteContactsEnable,
            siteContactData
        } = this.props
        const {
            isContactTypesPresent,
            contactTypes
        } = this.state;
        return (
            
            <Fragment>
                <FormHeadingWrapper>
                    <FormHeading><h1>{`${rootNode.partner.values.partnerName} site contacts`}</h1></FormHeading>
                    <Button className='edit-btn' disabled={iseditSiteContactsEnable || !isContactTypesPresent} onClick={() => editSiteContacts(true)} variant='secondary'>Edit site contacts </Button>
                </FormHeadingWrapper>
                <AdministrativeContactCard 
                            iseditSiteContactsEnable={iseditSiteContactsEnable}
                            editSiteContacts = {editSiteContacts}
                            contacts ={contacts}
                            rootNode = {rootNode}
                            node = {node}
                            siteContactData = {siteContactData}
                            updateEditableStatus = {(status) => this.updateEditableStatus(status)}
                        />
                {contactList.map((c, index) => {
                    let contactType = this.state.contactTypes[c.id];
                    return (
                        <>                        
                        <StyleGrayWrapper key={index}>
                        <CardTopContainer>
                            <Label dark={true} emphasized={true}>{c.label}</Label>
                            <Button disabled={iseditSiteContactsEnable} variant="secondary">
                                {i18Data.view}
                            </Button>
                        </CardTopContainer>
                        <CardBottomContainer>
                            {
                                (contactType && contactType.contactUuid) &&
                                <>                                      
                                    <StyledLeftSection spaceBottom>
                                    <Label dark={true} emphasized={true} >{!iseditSiteContactsEnable ? i18Data.contactName : i18Data.currentContact}</Label>
                                    <Label className='contact-name'>
                                        {contactType && `${contactType.firstName},${contactType.lastName}`}
                                        {iseditSiteContactsEnable && <IconStyle onClick={() => this.updateContact(c)}><Icon  className='close'  graphic="close" size="xs"/></IconStyle>}
                                    </Label>
                                    </StyledLeftSection>
                                    {!iseditSiteContactsEnable &&
                                        <StyledRightSection>
                                            <Label dark={true} emphasized={true}>{i18Data.status}</Label>
                                            <Label>
                                                {
                                                    <StatusIndicator variant={contactType && contactType.contactStatus === i18Data.active ? "success" : "error"}>
                                                        {contactType && contactType.contactStatus || i18Data.inActive}
                                                    </StatusIndicator>
                                                }
                                            </Label>
                                        </StyledRightSection>
                                    }
                                                                                                            
                                </>
                            }
                            {
                                (contactType && !contactType.contactUuid || iseditSiteContactsEnable)&&
                                <DropdownWrapper>
                                    <Label dark={true} emphasized={true}>
                                        {iseditSiteContactsEnable ? 'Assign a different contact' : 'Assign contact'}
                                    </Label>
                                    <Dropdown
                                        className='drop-down'
                                        onChange={(e) => this.onChange(c, e)}
                                        defaultValue='select'
                                        id={index}
                                        value={contactType.uuid || 'select'}
                                    >
                                        <option value="select">Select contact</option>
                                        {contacts && contacts.map((contact, index) => {
                                            return (
                                                <Fragment key={index}>
                                                    <option value={contact.contactUuid} id={contact.contactUuid} index={index}>
                                                        {`${contact.firstName},${contact.lastName}`}
                                                    </option>
                                                </Fragment>
                                            )
                                        })}
                                    </Dropdown>
                                </DropdownWrapper>
                            }

                        </CardBottomContainer>
                    </StyleGrayWrapper>
                    </>
                    )
                }

                )}
            </Fragment>)
    }
}
