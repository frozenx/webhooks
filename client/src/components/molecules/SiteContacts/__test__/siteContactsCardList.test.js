import React from 'react';
import SiteContactsCardList from '../SiteContactsCardList'
import renderer from 'react-test-renderer';
import { DefaultThemeProvider } from '@beans/theme';
import {shallow} from 'enzyme';
import ContactBook from '../../../pages/SupplierHeader/contactBook'


describe('SiteContacts', () => {
    let props,
        tree
    let contacts = [
        {
          "contactUuid": "774477537246487",
          "contactStatus": "Active",
          "firstName": "Volkswagen",
          "lastName": "Zetta",
        },
        {
          "contactUuid": "600853397549052",
          "contactStatus": "Active",
          "firstName": "Skoda",
          "lastName": "Rapid",          
        }
      ]    
   
    let siteContactList = [
        { id: 'billToContact', label: 'BillTo Contact'},
        { id: 'shipToContact', label: 'ShipTo Contact'},
        { id: 'correspondenceContact', label: 'Correspondence Contact'},
        { id: 'dunningContact', label: 'Dunning Contact'},
        { id: 'InvoicesContact', label: 'Invoices Contact'},
      ]
    beforeEach(() => {
        props = {
            contactList: siteContactList,
            siteContactData: jest.fn(),
            editSiteContacts: jest.fn(),
            contacts: contacts,            
            
            node: {
                contact: {
                    values: {
                        "billToContact": "774477537246487",
                        "shipToContact": "",
                        "correspondenceContact": "",
                        "dunningContact": "",
                        "InvoicesContact": "",
                        "administrativeContact": "774477537246487"
                    }
                }
            },
                        
            rootNode: {
                uuid: 1234,
                contactBook: new ContactBook(contacts),                
                partner: {
                    values: {
                        partnerName: 'TEST NAME'                    
                    }
                }
            }

        }

        tree = renderer.create(
            <DefaultThemeProvider>
                <SiteContactsCardList  {...props}/>
            </DefaultThemeProvider>
        ).toJSON();
    });
    test('should render the component', () => {
        expect(tree).toMatchSnapshot();
    });
    test('should call getContact method on componentDidMount', () => {
        const Wrapper = shallow(<SiteContactsCardList {...props}/>);            
        const instance = Wrapper.instance();
        jest.spyOn(instance, 'getContact');
        instance.componentDidMount();
        expect(instance.getContact).toHaveBeenCalledWith(props.node.contact.values.billToContact);
    });
    test('should call the siteContactData of props method when changed or selected the addressType', () => {
        const mockEvent = { target: { value : '774477537246487'} };
        const Wrapper = shallow(<SiteContactsCardList {...props}/>);
        Wrapper.find('.drop-down').first().simulate('change', mockEvent)
        expect(props.siteContactData).toHaveBeenCalled()
        expect(props.editSiteContacts).toHaveBeenCalled()
        expect(Wrapper.state().isContactTypesPresent).toBe(true)
    }); 
    
    test('should not populate any values for contact types if there are no previously saved contactTypes', () => {
        const node = {
            contact: {
                values: {
                    "billToContact": "",
                    "shipToContact": "",
                    "correspondenceContact": "",
                    "dunningContact": "",
                    "InvoicesContact": "",
                    "administrativeContact": ""
                }
            }
        }
        const mockProps = {
            ...props, node
        }
        const Wrapper = shallow(<SiteContactsCardList {...mockProps}/>);            
        expect(Wrapper.find('.contact-name')).toHaveLength(0);
    });

    test('should not return any contact if the contact is not found in the contactBook', () => {
        const node = {
            contact: {
                values: {
                    "billToContact": "1234567",
                    "shipToContact": "",
                    "correspondenceContact": "",
                    "dunningContact": "",
                    "InvoicesContact": "",
                    "administrativeContact": ""
                }
            }
        }
        const mockProps = {
            ...props, node            
        }
        const Wrapper = shallow(<SiteContactsCardList {...mockProps}/>);            
        expect(Wrapper.find('.contact-name')).toHaveLength(0);
    });
    describe('when iseditSiteContactsEnable is true', () => {
        let Wrapper;
        beforeEach(() => {
            Wrapper = shallow(<SiteContactsCardList {...props} iseditSiteContactsEnable={true}/>);
        })
        test('should show the close icon if the iseditSiteContactsEnable prop is true', () => {
            expect(Wrapper.find('.close')).toHaveLength(1)
        });
        test('should call siteContactData method of props if the close icon is clicked', () => {
            Wrapper.find('.close').simulate('click')
            expect(props.siteContactData).toHaveBeenCalled()
        });
        test('should call editSiteContacts method of props if the edit button is clicked', () => {
            Wrapper.find('.edit-btn').simulate('click')
            expect(props.editSiteContacts).toHaveBeenCalled()
        });
        test('should set the state isContactTypesPresent if updateEditableStatus method is called', () => {
            let status = true
            const instance = Wrapper.instance();
            jest.spyOn(instance, 'updateEditableStatus')
            instance.updateEditableStatus(status)
            expect(Wrapper.state().isContactTypesPresent).toBe(status);
        });
    })   
});