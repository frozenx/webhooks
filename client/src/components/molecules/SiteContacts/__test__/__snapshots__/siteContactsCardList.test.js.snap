// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteContacts should render the component 1`] = `
Array [
  <div
    className="base-components__RootElement-sc-150tvch-1 sc-chPdSV kkjGzd"
  >
    <div
      className="base-components__RootElement-sc-150tvch-1 sc-EHOje ebEHwY"
    >
      <h1>
        TEST NAME site contacts
      </h1>
    </div>
  </div>,
  <div
    className="base-components__RootElement-sc-150tvch-1 sc-kAzzGY bphYLh"
  >
    <div
      className="sc-kgoBCf ffTmed"
    />
    <div
      className="sc-kGXeez kScwyN"
    >
      <div
        className="sc-kpOJdX dPFWje"
      />
      <div
        className="sc-kpOJdX sc-dxgOiQ gHISXr"
      />
      <div
        className="sc-eNQAEJ kECEEu"
      >
        <div
          className="base-components__RootElement-sc-150tvch-1 cCmuza"
        >
          <select
            className="base-components__BaseInput-sc-150tvch-2 beans-dropdown__select styled__Select-sc-10jppqs-1 GLMMI"
            disabled={false}
            onChange={[Function]}
            role="listbox"
            value="select"
          >
            <option
              value="select"
            >
              Select contact
            </option>
            <option
              id="600853397549052"
              index={0}
              value="600853397549052"
            >
              Skoda,Rapid
            </option>
          </select>
          <svg
            aria-hidden={true}
            className="styled__AlignedIcon-sc-10jppqs-0 hHyXUl styled__StyledSVG-sc-1nulmu6-1 flhEYZ"
            color="currentColor"
            disabled={false}
            height={12}
            preserveAspectRatio="xMinYMax meet"
            viewBox="0 -1 20 12"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.652 1.312L10 10.276 1.356 1.312"
              fill="none"
              stroke="#666666"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="base-components__RootElement-sc-150tvch-1 sc-kAzzGY bphYLh"
  >
    <div
      className="sc-kgoBCf ffTmed"
    />
    <div
      className="sc-kGXeez kScwyN"
    >
      <div
        className="sc-kpOJdX dPFWje"
      />
      <div
        className="sc-kpOJdX sc-dxgOiQ gHISXr"
      />
    </div>
  </div>,
  <div
    className="base-components__RootElement-sc-150tvch-1 sc-kAzzGY bphYLh"
  >
    <div
      className="sc-kgoBCf ffTmed"
    />
    <div
      className="sc-kGXeez kScwyN"
    >
      <div
        className="sc-eNQAEJ kECEEu"
      >
        <div
          className="drop-down base-components__RootElement-sc-150tvch-1 cCmuza"
        >
          <select
            className="base-components__BaseInput-sc-150tvch-2 beans-dropdown__select styled__Select-sc-10jppqs-1 GLMMI"
            disabled={false}
            id={1}
            onChange={[Function]}
            role="listbox"
            value="select"
          >
            <option
              value="select"
            >
              Select contact
            </option>
            <option
              id="774477537246487"
              index={0}
              value="774477537246487"
            >
              Volkswagen,Zetta
            </option>
            <option
              id="600853397549052"
              index={1}
              value="600853397549052"
            >
              Skoda,Rapid
            </option>
          </select>
          <svg
            aria-hidden={true}
            className="styled__AlignedIcon-sc-10jppqs-0 hHyXUl styled__StyledSVG-sc-1nulmu6-1 flhEYZ"
            color="currentColor"
            disabled={false}
            height={12}
            preserveAspectRatio="xMinYMax meet"
            viewBox="0 -1 20 12"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.652 1.312L10 10.276 1.356 1.312"
              fill="none"
              stroke="#666666"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="base-components__RootElement-sc-150tvch-1 sc-kAzzGY bphYLh"
  >
    <div
      className="sc-kgoBCf ffTmed"
    />
    <div
      className="sc-kGXeez kScwyN"
    >
      <div
        className="sc-eNQAEJ kECEEu"
      >
        <div
          className="drop-down base-components__RootElement-sc-150tvch-1 cCmuza"
        >
          <select
            className="base-components__BaseInput-sc-150tvch-2 beans-dropdown__select styled__Select-sc-10jppqs-1 GLMMI"
            disabled={false}
            id={2}
            onChange={[Function]}
            role="listbox"
            value="select"
          >
            <option
              value="select"
            >
              Select contact
            </option>
            <option
              id="774477537246487"
              index={0}
              value="774477537246487"
            >
              Volkswagen,Zetta
            </option>
            <option
              id="600853397549052"
              index={1}
              value="600853397549052"
            >
              Skoda,Rapid
            </option>
          </select>
          <svg
            aria-hidden={true}
            className="styled__AlignedIcon-sc-10jppqs-0 hHyXUl styled__StyledSVG-sc-1nulmu6-1 flhEYZ"
            color="currentColor"
            disabled={false}
            height={12}
            preserveAspectRatio="xMinYMax meet"
            viewBox="0 -1 20 12"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.652 1.312L10 10.276 1.356 1.312"
              fill="none"
              stroke="#666666"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="base-components__RootElement-sc-150tvch-1 sc-kAzzGY bphYLh"
  >
    <div
      className="sc-kgoBCf ffTmed"
    />
    <div
      className="sc-kGXeez kScwyN"
    >
      <div
        className="sc-eNQAEJ kECEEu"
      >
        <div
          className="drop-down base-components__RootElement-sc-150tvch-1 cCmuza"
        >
          <select
            className="base-components__BaseInput-sc-150tvch-2 beans-dropdown__select styled__Select-sc-10jppqs-1 GLMMI"
            disabled={false}
            id={3}
            onChange={[Function]}
            role="listbox"
            value="select"
          >
            <option
              value="select"
            >
              Select contact
            </option>
            <option
              id="774477537246487"
              index={0}
              value="774477537246487"
            >
              Volkswagen,Zetta
            </option>
            <option
              id="600853397549052"
              index={1}
              value="600853397549052"
            >
              Skoda,Rapid
            </option>
          </select>
          <svg
            aria-hidden={true}
            className="styled__AlignedIcon-sc-10jppqs-0 hHyXUl styled__StyledSVG-sc-1nulmu6-1 flhEYZ"
            color="currentColor"
            disabled={false}
            height={12}
            preserveAspectRatio="xMinYMax meet"
            viewBox="0 -1 20 12"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.652 1.312L10 10.276 1.356 1.312"
              fill="none"
              stroke="#666666"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="base-components__RootElement-sc-150tvch-1 sc-kAzzGY bphYLh"
  >
    <div
      className="sc-kgoBCf ffTmed"
    />
    <div
      className="sc-kGXeez kScwyN"
    >
      <div
        className="sc-eNQAEJ kECEEu"
      >
        <div
          className="drop-down base-components__RootElement-sc-150tvch-1 cCmuza"
        >
          <select
            className="base-components__BaseInput-sc-150tvch-2 beans-dropdown__select styled__Select-sc-10jppqs-1 GLMMI"
            disabled={false}
            id={4}
            onChange={[Function]}
            role="listbox"
            value="select"
          >
            <option
              value="select"
            >
              Select contact
            </option>
            <option
              id="774477537246487"
              index={0}
              value="774477537246487"
            >
              Volkswagen,Zetta
            </option>
            <option
              id="600853397549052"
              index={1}
              value="600853397549052"
            >
              Skoda,Rapid
            </option>
          </select>
          <svg
            aria-hidden={true}
            className="styled__AlignedIcon-sc-10jppqs-0 hHyXUl styled__StyledSVG-sc-1nulmu6-1 flhEYZ"
            color="currentColor"
            disabled={false}
            height={12}
            preserveAspectRatio="xMinYMax meet"
            viewBox="0 -1 20 12"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.652 1.312L10 10.276 1.356 1.312"
              fill="none"
              stroke="#666666"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
]
`;
